docs/en/operations/server-configuration-parameters/settings.md

```diff
@@ -1628,6 +1628,45 @@ Default value: `0.5`.
 
 
 
+## async_load_databases {#async_load_databases}
+
+Asynchronous loading of databases and tables.
+
+If `true` all non-system databases with `Ordinary`, `Atomic` and `Replicated` engine will be loaded asynchronously after the ClickHouse server start up. See `system.async_loader` table, `async_loader_background_pool_size` and `async_loader_foreground_pool_size` server settings. Any query that tries to access a table, that is not yet loaded, will wait for exactly this table to be started up. If load job fails, query will rethrow an error (instead of shutting down the whole server in case of `async_load_databases = false`). The table that is waited for by at least one query will be loaded with higher priority. DDL queries on a database will wait for exactly that database to be started up.
+
+If `false`, all databases are loaded when the server starts.
+
+The default is `false`.
+
+**Example**
+
+``` xml
+<async_load_databases>true</async_load_databases>
+```
+
+## async_loader_foreground_pool_size {#async_loader_foreground_pool_size}
+
+Sets the number of threads performing asynchronous load jobs in foreground pool. The foreground pool is used for loading table synchronously before server start listening on a port and for loading tables that are waited for. Foreground pool has higher priority than background pool. It means that no job starts in background pool while there are jobs running in foreground pool.
+
+Possible values:
+
+-   Any positive integer.
+-   Zero. Use all available CPUs.
+
+Default value: 0.
+
+
+## async_loader_background_pool_size {#async_loader_background_pool_size}
+
+Sets the number of threads performing asynchronous load jobs in background pool. The background pool is used for loading tables asynchronously after server start in case there are no queries waiting for the table. It could be beneficial to keep low number of threads in background pool if there are a lot of tables. It will reserve CPU resources for concurrent query execution.
+
+Possible values:
+
+-   Any positive integer.
+-   Zero. Use all available CPUs.
+
+Default value: 16.
```

Comment:
Yeah, 0 looks better, everybody could lower it if he wants to leave space for query execution while background loading.

Reply:
Yeah, 0 looks better, everybody could lower it if he wants to leave space for query execution while background loading.

Reply:
Also, we can use half of available CPUs for 0 (so it will leave space for query execution by default, and will dynamically choose a reasonable number of threads in current env)

Reply:
Now I'm not sure it is a good idea because with such a default, the server will start longer (I mean load all tables; port will open faster of course) after enabling the feature.

Reply:
We discussed it and decided to leave "0 means all the cpus" and leave OS to deal with the excess number of threads. Default fixed to be zero.